samanthas_telegram_bot/api_clients/base/base_api_client.py

```diff
@@ -23,6 +23,11 @@
 
 
 class BaseApiClient:
+    """Class for making API requests from within the conversation with the user.
+
+    All public methods require update and context to be passed.
```

Comment:
This seems like a bad idea to me, because it breaks single responsibility. Api caller should not know anything about update and context of the telegram bot

Reply:
In a general case, I agree and I try to follow the approach you're describing. But here everything is about the bot, and the knowledge that there is a bot application that has an update and a context doesn't seem like too much knowledge to me. It's sort of "everywhere" in the whole project.

Reply:
I slightly changed the philosophy, but most methods still require update and context and I still think it's OK in this case.